File: ArrowMove/src/game/data/TextCommandData.java
package game.data;

import game.controller.Controller;
import game.system.Game;

public class TextCommandData extends CommandData {
	private String text;
	
	public TextCommandData(Controller controller,String text) {
		super(controller);
		this.text = text;
	}
	public String toString() {
		return text;
	};
	
	@Override
	public TextCommandData toTextCommandData() {
		return new TextCommandData(controller, text);
	}
	@Override
	public void execute(Game game) {
		game.makeCommandData(this).execute(game);
	}
	@Override
	public void undo(Game game) {
		game.makeCommandData(this).undo(game);
		
	}
}

File: main/java/game/data/TextCommandData.kt
package game.data

import game.controller.Controller
import game.system.Game

class TextCommandData(controller: Controller<*>, private val text: String) : CommandData(controller) {
    override fun toString(): String {
        return text
    }

    @Override
    override fun toTextCommandData(): TextCommandData {
        return TextCommandData(controller, text)
    }
//
//    @Override
//    fun execute(game: Game<*>) {
//        game.makeCommandData(this).execute(game)
//    }
//
//    @Override
//    fun undo(game: Game<*>) {
//        game.makeCommandData(this).undo(game)
//
//    }
}
